t/parser.t

```diff
@@ -34,6 +38,34 @@ is_a_flt   = 4.2
     ok !ref($data->{not_a_flt}), 'strings do not inflate as floats';
     ok ref($data->{is_a_flt}) && $data->{is_a_flt}->isa('Math::BigFloat'), 'floats do inflate with inflate_float';
   };
+
+  subtest 'no dequoting of raw string keys' => sub {
+    my $funky_string = "a \\\\ \\x42 b \\\"";
+    diag 'RAW FUNKY STRING';
+    diag $funky_string;
+    diag '';
+    my $toml = <<END;
+key.'$funky_string' = 'a'
+itable = { '$funky_string' = 'b' }
+[table.'$funky_string']
+x = 'c'
+[[array.'$funky_string']]
+y = 'd'
+END
+    diag 'INPUT TOML';
+    diag $toml;
+    diag '';
+    my $data = eval { from_toml($toml) };
+    my $error = $@;
+    ok !$error, "parsed ok" or diag $error;
+    diag 'PARSED AS';
+    diag Dumper($data);
+    diag '';
+    ok $data->{key}{$funky_string} eq 'a', "key.funky";
+    ok $data->{itable}{$funky_string} eq 'b', "not in inline table";
+    ok $data->{table}{$funky_string}{x} eq 'c', "not in table heading";
+    ok $data->{array}{$funky_string}[0]{y} eq 'd', "not in array item heading";
+  };
```

Comment:
Thanks for the extra tests. Could we make the diagnostic output optional? That would keep the default test output more succinct.

Reply:
Done.  I'm not sure I've done it in the best way, though.

Reply:
Seems fixed now. 👍🏻